blog/Numtrip_blog.md

```diff
@@ -30,3 +31,73 @@ def auswerten(zeile, spalte):
 ```
 
 - Als letztes habe ich alle diese Funktionen in eine gesamte Funktion "play()" hineingeschrieben und diesen Vorgang solange das Spiel läuft (momentan unendlich lange) laufen lassen.
+
+### **Validierung: (2.12.22)**
+
+- Ich habe mit Alice sehr viel besprochen. Bei mir waren Probleme beim Validieren aufgetreten, da ich die Werte falsch oder als String übergeben habe. Bei ihr wurde das Wort "True" übergeben. Beides konten wir jedoch beheben.
+
+```py
+def eingabe_validieren(zeile, spalte):
+    try:
+        zeile = int(zeile) - 1
+        spalte = int(spalte) - 1
+        return True
+    except:
+        print('Zeilen- und Spaltennummer müssen Zahlen sein!')
+        return False
+```
+
+??? Aufgetauchte Fragen: Muss hier 'return True' geschrieben werden? Wird hier zeile und spalte übergeben?
+
+```py
+def eingabe():
+    zeile = " "
+    spalte = " "
+    while not eingabe_validieren(zeile, spalte):
+        zeile = input('Gib eine Zeilennummer zwischen 1 und 6 ein:')
+        spalte = input('Gib eine Spaltennummer zwischen 1 und 6 ein:')
+    return (int(zeile), int(spalte))
+```
+
+!!! WICHTIG: Die Zahlen müssen noch einmal mit int() in Zahlen verwandelt werden, weil bei der Validierung bloss geschaut wird, ob es möglich wäre, diese in Zahlen umzuwandeln.
+
+### **Nachbarzellen kombinieren**
+
+- Ich habe versucht den Floodfill-code so umzuwandeln, dass er in mein Spiel passt.
+
+??? Leider ist ein kleines Problem aufgetreten. Da ich die Zahl bereits vorher auf 0 gesetzt hatte und es also testet, ob diese Zahl (also 0) eine 2 ist - was ja nie der Fall sein wird.
+
+```py
+def nachbarfelder_löschen(x, y, vorherige_Zahl):
+    if spielfeld[x][y] == vorherige_Zahl:  # Zahl ist ja bereits 0??
+        spielfeld[x][y] = 0
+        nachbarfelder_löschen(x, y + 1)  # rechts
+        nachbarfelder_löschen(x, y - 1)  # links
+        nachbarfelder_löschen(x + 1, y)  # unten
+        nachbarfelder_löschen(x - 1, y)  # oben
+```
+
+???Ausserdem wird nur die Zahl 2 getestet und nicht die Zahl, die vorher im Kasten stand - ich weiss nicht wie ich diese übergeben soll.
+
+```py
+nachbarfelder_löschen(zeile, spalte, 2)
+```
+
+- Die Probleme habe ich nun behoben, indem ich die Funktion auswerten ganz weggenommen habe (und somit die Zahl erst im "nachbarfelder_kombinieren" auf 0 gesetzt wird.) und die Funktion "nachbarfelder_komninieren" als Ganzes in "Auswertung()" umgewandelt habe. Zusätzlich habe ich eine Zahl vorher übergeben, welche die Zahl in diesem Feld speichert und bei der rekursiven Auswerten-funktion benutzt werden kann. Damit es keine Fehler gibt, habe ich die Rahmenbedingung in den Code eingebaut.
```

Comment:
Wenn Sie über Code oder Funktionsaufrufe in einer Zeile sprechen, können Sie diese Teile mit einfachen "Backticks" markieren:

 ```suggestion
- Die Probleme habe ich nun behoben, indem ich die Funktion auswerten ganz weggenommen habe (und somit die Zahl erst im `nachbarfelder_kombinieren` auf 0 gesetzt wird.) und die Funktion `nachbarfelder_kombinieren` als Ganzes in `Auswertung()` umgewandelt habe. Zusätzlich habe ich eine Zahl vorher übergeben, welche die Zahl in diesem Feld speichert und bei der rekursiven Auswerten-funktion benutzt werden kann. Damit es keine Fehler gibt, habe ich die Rahmenbedingung in den Code eingebaut.
```